server/src/services/animal-location.service.ts

```diff
@@ -89,19 +89,16 @@ export class AnimalLocationService {
         });
     }
 
-    getAllAnimalsLocationTableData(): any {
-
-        const con =  this.databaseService.getConnection();
-        return con.then(async (data)=>{
-           return await data.getRepository(AnimalLocation).find(); 
-        })  
+    async getAllAnimalsLocationTableData(): Promise<JSON> {
+        const con = await this.databaseService.getConnection();
+        let animaldata =  JSON.parse(JSON.stringify( await con.getRepository(AnimalLocation).find()));
+        return animaldata;
     }
 
-    getIndividualAnimalLocationTableData(animalID): any {
-
-        const con =  this.databaseService.getConnection();
-        return con.then(async (data)=>{
-           return await con.getRepository(AnimalLocation).find({ animalId : animalID }); 
+    async getIndividualAnimalLocationTableData(animalID): Promise<JSON> {
+        const con = await this.databaseService.getConnection();
+        let animaldata =  JSON.parse(JSON.stringify( await con.getRepository(AnimalLocation).find({ animalId : animalID })));
```

Comment:
Why do you need to call JSON.parse(JSON.stringify(...)) here? Do we need a deep copy?

Reply:
No we do not. was for testing purposes, overlooked it when pushed.